### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by samisuteria**
add swiftlint and fix critical errors

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+disabled_rules: # rule identifiers to exclude from running
+  - trailing_whitespace
```
Comment (compnerd) [inline .swiftlint.yml]: Why do we want to disable this rule?

File: WalkingChallenge/Facebook.swift [hunk 2]
```diff
@@ -38,8 +38,8 @@ class Facebook {
       return nil
     }
 
-    return Friend(fbid: id, display_name: name, first_name: first_name,
-                  last_name: last_name, picture_url: picture_url)
+        return Friend(fbid: id, displayName: name, firstName: first_name,
```
Comment (compnerd) [inline WalkingChallenge/Facebook.swift]: Indentation is wrong here.

File: WalkingChallenge/Leaderboard.swift [hunk 3]
```diff
@@ -120,7 +120,12 @@ extension LeaderBoard: UITableViewDataSource {
...
 
     cell.configure(entry)
-    return cell as! UITableViewCell
+    // TODO: Sami fix this to use actual CellInfo pattern.
+    if let cell = cell as? UITableViewCell {
+      return cell
+    } else {
+      return UITableViewCell()
+    }
   }
```
Comment (compnerd) [inline WalkingChallenge/Leaderboard.swift]: Cant this be written as:

    return cell as? UITableViewCell ?? UITableViewCell()

**Review by compnerd — APPROVED** [review]

File: .swiftlint.yml (hunk 1 above)
Comment (samisuteria) [reply .swiftlint.yml]: I personally like trailing whitespace between functions which this rule gives a warning for

Comment (compnerd) [reply .swiftlint.yml]: Whitespace between functions is fine, I don't see why the trailing whitespace is needed though.

Comment (samisuteria) [reply .swiftlint.yml]: That rule was being triggered for whitespace between functions

File: WalkingChallenge/Facebook.swift [hunk 4]
```diff
@@ -38,7 +38,7 @@ class Facebook {
       return nil
     }
 
-        return Friend(fbid: id, displayName: name, firstName: first_name,
+    return Friend(fbid: id, displayName: name, firstName: first_name,
                       lastName: last_name, pictureRawURL: picture_url)
```
Comment (compnerd) [inline WalkingChallenge/Facebook.swift]: Indentation is still off on this line :-(.

**Comment by compnerd** [comment]
Thanks!  This should be good to ensure that overall the style matches :-)

**Pull request merged**